oauth2/src/test/groovy/com/okta/spring/boot/oauth/env/OktaEnvironmentPostProcessorApplicationListenerTest.groovy

```diff
@@ -0,0 +1,119 @@
+import org.springframework.mock.env.MockEnvironment
+import org.testng.annotations.AfterMethod
+import org.testng.annotations.Test
+
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.is
+import static org.mockito.Mockito.mock
+import static org.mockito.Mockito.when
+
+class OktaEnvironmentPostProcessorApplicationListenerTest {
+
+    private static final String LOGGER_NAME = OktaEnvironmentPostProcessorApplicationListener.getName()
+    private static final String MSG = "Your Okta Issuer URL is missing. You can copy your domain from the Okta Developer Console. " +
+        "Follow these instructions to find it: https://bit.ly/finding-okta-domain" + System.lineSeparator() +
+        "To fix this add the `okta.oauth2.issuer` property to your application environments."
+    private static final Logger log = (Logger) LoggerFactory.getLogger(LOGGER_NAME)
+
+    def event = mock(ApplicationPreparedEvent)
+    def context = mock(ConfigurableApplicationContext)
+
+    @AfterMethod
+    void cleanUp() {
+        log.detachAndStopAllAppenders()
+    }
+
+    @Test
+    void testValidIssuerProperty() {
+        LogsAppender logsAppender = new LogsAppender()
+        logsAppender.start()
+        log.addAppender(logsAppender)
+
+        buildApplicationEvent([
+            "okta.oauth2.issuer": "https://issuer.example.com/foobar"
+        ])
+        new OktaEnvironmentPostProcessorApplicationListener().onApplicationEvent(event)
+
+        def logs = logsAppender.getLogs(LOGGER_NAME)
+
+        assertThat "The logs list must be empty", logs.isEmpty(), is(true)
+        assertThat "Should be no WARN messages", logsAppender.contains(Level.WARN), is(false)
+    }
+
+    @Test
+    void testEmptyIssuer() {
+        LogsAppender logsAppender = new LogsAppender()
+        logsAppender.start()
+        log.addAppender(logsAppender)
+
+        buildApplicationEvent(Collections.emptyMap())
+        new OktaEnvironmentPostProcessorApplicationListener().onApplicationEvent(event)
+
+        def logs = logsAppender.getLogs(LOGGER_NAME)
+
+        assertThat "Should be at least one WARN message", logsAppender.contains(Level.WARN), is(true)
+        assertThat "Should be one message", logs.size(), is(1)
+        assertThat "Wrong level", logs.get(0).getLevel(), is(Level.WARN)
+        assertThat "Wrong message", logs.get(0).getMessage(), is(MSG)
+    }
+
+    private void buildApplicationEvent(Map<String, Object> properties) {
+        def environment = new MockEnvironment()
+        environment.getPropertySources().addFirst(new MapPropertySource("test", properties))
+        when(context.getEnvironment()).thenReturn(environment)
+        when(event.getApplicationContext()).thenReturn(context)
+    }
+
+    private static final class LogsAppender extends ListAppender<ILoggingEvent> {
+        private final String threadName
+
+        LogsAppender() {
+            threadName = Thread.currentThread().getName()
```

Comment:
nit: The thread name isn't guaranteed to be unique

Reply:
Changed to `thread = Thread.currentThread()`